analysis_scripts/qc_scripts/bg_nature_qc.Rmd

```diff
@@ -0,0 +1,229 @@
+    
+    json_res = fromJSON(curr_file)
+
+    # format authors
+    authors = unlist(lapply(json_res$authors, function(x) paste(unlist(x$name), collapse="; ")))
+
+    # make df
+    authors_df = data.frame(file_id=json_res$file_id,
+                            year=json_res$year,
+                            authors=authors)
+    
+    all_authors = rbind(all_authors, authors_df)
+
+}
+
+all_authors = all_authors[-1,]
+
+# format file_id into a doi
+all_authors$doi = paste("doi:10.1038/", all_authors$file_id, sep="")
+
+# plot number of articles scraped
+ggplot(unique(all_authors[,c("file_id", "year")]), aes(x=as.factor(year))) +
+    geom_bar() + theme_bw() +
+    xlab("Year of Article") + ylab("# articles") +
+        ggtitle("# Nature Research Articles + Letters Over Time")
+
+```
+
+
+
+## Pipeline Step 2: Gender Predictions
+
+### Check if gender predictions were done on all scraped data
+
+```{r step2_analyze, echo=TRUE, warning=FALSE, message=F}
+
+
+# read in the scraped nature data
+pipeline_2_file = file.path(proj_dir,
+                    "/data/author_data/nature_author_gender.tsv")
+gender_res = fread(pipeline_2_file)
+
+authored_df = all_authors
+
+# files scraped but have no gender prediction
+gender_missing = setdiff(unique(all_authors$doi), unique(gender_res$doi))
+authored_df$no_gender = FALSE
+authored_df$no_gender[which(authored_df$doi %in% gender_missing)] = TRUE
+print(paste("% of DOIs with no gender prediction:", 
+            length(gender_missing)/length(unique(authored_df$doi))))
+
+
+# plot number of springer articles with no gender prediction
+ggplot(unique(authored_df[,c("doi", "year", "no_gender")]), 
+       aes(x=as.factor(year), fill=no_gender)) +
+        geom_bar(position="fill") + theme_bw() +
+        xlab("Year of Article") + ylab("% nature articles with no gender prediction") +
+            ggtitle("% nature articles with no gender prediction")
+
+
+# single author publications are ignored, so remove them
+authored_df = unique(authored_df)
+no_gender_authored_df = subset(authored_df, no_gender == TRUE)
+num_author = lapply(no_gender_authored_df$authors, function(x) length(grep(";", x))+1)
+no_gender_authored_df = no_gender_authored_df[which(num_author > 1),]
+print(paste("% of DOIs with no gender prediction after filtering single author pubs:", 
+            nrow(no_gender_authored_df)/length(unique(authored_df$doi))))
+
+# plot number of Nature articles with no gender prediction
+ggplot(unique(no_gender_authored_df[,c("doi", "year")]), 
+       aes(x=as.factor(year))) +
+        geom_bar() + theme_bw() +
+        xlab("Year of Article") + ylab("% Nature articles with no gender prediction") +
+            ggtitle("% Nature articles with no gender prediction after filtering for multi-author")
+
+# now the remaining should all be abreviated first names
+first_authors = unlist(lapply(no_gender_authored_df$authors, function(x) unlist(str_split(x, "; "))[1]))
+print(head(first_authors))
```

Comment:
This is a very minor suggestion. What I usually do when things that I expect to happen is to add `assert` (in Python) or `stopifnot` in R. So in case you have to run this notebook again for any reason, it is "autochecked" and it saves you a little bit from manually inspecting outputs.

Reply:
added in: https://github.com/greenelab/nature_news_disparities/pull/19/commits/d2f65cfa1ebc94522281e63ab529fe48c0cf1bb4